src/projections.jl

```diff
@@ -152,6 +158,116 @@ function vec_symm(X)
     return X[LinearAlgebra.triu(trues(size(X)))]
 end
 
+"""
+    projection_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.ExponentialCone) where {T}
+
+projection of vector `v` on closure of the exponential cone
+i.e. `cl(Kexp) = {(x,y,z) | y e^(x/y) <= z, y>0 } U {(x,y,z)| x <= 0, y = 0, z >= 0}`.
+
+References:
+* [Proximal Algorithms, 6.3.4](https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf)
+by Neal Parikh and Stephen Boyd.
+* [Projection, presolve in MOSEK: exponential, and power cones]
+(https://docs.mosek.com/slides/2018/ismp2018/ismp-friberg.pdf) by Henrik Friberg
+"""
+function projection_on_set(::DefaultDistance, v::AbstractVector{T}, s::MOI.ExponentialCone) where {T}
+    _check_dimension(v, s)
+
+    if _in_exp_cone(v; dual=false)
+        return v
+    elseif _in_exp_cone(-v; dual=true)
+        # if in polar cone Ko = -K*
+        return zeros(3)
+    elseif v[1] <= 0 && v[2] <= 0
+        return [v[1]; 0.0; max(v[3],0)]
+    else
+        return _exp_cone_proj_case_4(v)
+    end
+end
+
+function _in_exp_cone(v::AbstractVector{T}; dual=false) where {T}
```

Comment:
EXP_CONE_THRESH should be a keyword arg to `_in_exp_cone` instead of a global constant.
Also, couldn't `_in_exp_cone` just use `distance_to_set(d, v, ::MOI.ExponentialCone)`?

Reply:
For numerical reasons, I define the cone as its closure with some tolerance for the boundary. I'm not sure if it makes sense for this to be in the `distance_to_set(d, v, ::MOI.ExponentialCone)` function as well.

Reply:
The current implementation of distance to set is doing (roughly) the same thing and also using the closure, even for the strict inequality on s / y so this will work in any case for testing whether the point is in the set

Reply:
Changed. Added `1e-8` tolerances in the `projection_on_set` function for the exponential cone and its dual.